nearcore/src/lib.rs

```diff
@@ -74,17 +71,87 @@ pub fn get_default_home() -> PathBuf {
     PathBuf::default()
 }
 
+fn db_checkpoint_root_dir<'a>(path: &'a Path, near_config: &'a NearConfig) -> &'a Path {
+    if let Some(db_checkpoints_path) = &near_config.config.db_checkpoints_path {
+        &db_checkpoints_path
+    } else {
+        path
+    }
+}
+
+const DB_CHECKPOINT_PREFIX: &str = "db_checkpoint_";
+
+fn find_db_checkpoint(path: &Path, near_config: &NearConfig) -> Result<Option<PathBuf>, io::Error> {
+    let db_checkpoints_path = db_checkpoint_root_dir(path, near_config);
+    for entry in std::fs::read_dir(db_checkpoints_path)?.filter(|entry| {
```

Comment:
Probably it makes sense to check here that exactly one snapshot exists and panic otherwise.

Reply:
I meant that we can check that no more than one snapshot exists otherwise it means the user did migration having another pending migration in a snapshot